pkg/report/report.go

```diff
@@ -614,3 +630,41 @@ func highestMatchRisk(mrs yara.MatchRules) int {
 	}
 	return highestRisk
 }
+
+// highestBehaviorRisk returns the highest risk score from a slice of FileReport Behaviors.
+func highestBehaviorRisk(fr malcontent.FileReport) int {
+	if len(fr.Behaviors) == 0 {
+		return 0
+	}
+
+	highestRisk := 0
+	for _, b := range fr.Behaviors {
+		if b.RiskScore > highestRisk {
+			highestRisk = b.RiskScore
+		}
+	}
+
+	return highestRisk
+}
+
+// handleOverrides modifies the behavior slice based on the contents of the override slice.
+func handleOverrides(fr malcontent.FileReport) []*malcontent.Behavior {
+	behaviorMap := make(map[string]*malcontent.Behavior, len(fr.Behaviors))
+	for _, b := range fr.Behaviors {
+		behaviorMap[b.RuleName] = b
+	}
+
+	for _, o := range fr.Overrides {
+		if b, exists := behaviorMap[o.Override]; exists {
```

Comment:
I tested this out with a critical false positive and it works.

Override rule:
```
rule pulumi_shellcode : override {
  meta:
    malware_shellcode_hash = "low"
    description = "Ignore shellcode findings in the pulumi binary"
  strings:
    $pulumi = { 50 75 6C 75 6D 69 }
    $pulumi2 = { 70 75 6C 75 6D 69 }
  condition:
    any of ($pulumi*)
}
```

Result:
![CleanShot 2024-10-04 at 13 16 39@2x](https://github.com/user-attachments/assets/13525b7f-da9f-4703-8676-190b8ed56776)